store/sqlstore/compliance_store.go

```diff
@@ -187,24 +212,34 @@ func (s SqlComplianceStore) ComplianceExport(job *model.Compliance) ([]*model.Co
 			Channels,
 			Users,
 			Posts
-		LEFT JOIN Bots ON Bots.UserId = Posts.UserId
+		LEFT JOIN
+			Bots ON Bots.UserId = Posts.UserId
 		WHERE
 			Channels.TeamId = ''
 				AND Posts.ChannelId = Channels.Id
 				AND Posts.UserId = Users.Id
-				AND Posts.CreateAt > :StartTime
+				AND (
+					Posts.CreateAt > :LastPostCreateAt
+					OR (Posts.CreateAt = :LastPostCreateAt AND Posts.Id > :LastPostId)
+				)
 				AND Posts.CreateAt <= :EndTime
 				` + emailQuery + `
-				` + keywordQuery + `)
-		ORDER BY PostCreateAt
-		LIMIT 30000`
-
-	var cposts []*model.CompliancePost
+				` + keywordQuery + `
+		ORDER BY Posts.CreateAt, Posts.Id
```

Comment:
@maxerenberg What's the performance like after adding `Posts.Id` to the `ORDER BY` clause?

Reply:
@mkraft
Here are the new results:

| # of posts | With keyset pag. | Without keyset pag. |
| ---------: | ----------------: | -------------------: |
| 30,000     | 1.594                    | 1.610                         |
| 300,000  |  16.599                 | 20.754                      |
| 3,000,000 | 336.325             |  934.949                   |

Although adding Posts.Id slowed down the keyset pagination considerably, it's still much faster than using OFFSET + LIMIT.

Reply:
Nice @maxerenberg! Results looks great. We can totally move ahead with this 👍🏻